src/relay/pass/let_list.h

```diff
@@ -0,0 +1,75 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file let_list.h
+ * \brief Let List Let you Let stuff yo.
+ */
+#ifndef TVM_RELAY_PASS_LET_LIST_H_
+#define TVM_RELAY_PASS_LET_LIST_H_
+
+#include <utility>
+#include <vector>
+#include <tuple>
+#include "tvm/relay/type.h"
+#include "incomplete_type.h"
+
+namespace tvm {
+namespace relay {
+
+struct PartialVar {
+  Var v;
+  explicit PartialVar(const Var & v) : v(v) { }
+  PartialVar() : v(VarNode::make("x")) { }
+};
+
+struct LetList {
+  std::vector<std::tuple<PartialVar, Type, Expr>> lets;
+  PartialVar let(const Type & ty, const Expr & expr) {
+    PartialVar id;
+    std::tuple<PartialVar, Type, Expr> tuple(id, ty, expr);
+    lets.push_back(tuple);
+    return id;
+  }
+
+  PartialVar let(const Expr & expr) {
+    return let(IncompleteTypeNode::make(TypeParamNode::kType), expr);
+  }
+
+  Expr plug(const Expr & expr) const {
+    Expr ret = expr;
+    for (auto rit = lets.rbegin(); rit != lets.rend(); rit++) {
+      ret = LetNode::make(std::get<0>(*rit).v, std::get<2>(*rit), ret, std::get<1>(*rit));
+    }
+    return ret;
+  }
+
+  static Expr with(std::function<Expr(LetList&)> f) {
+    LetList ll;
+    return ll.plug(f(ll));
+  }
+};
+
+struct ParamLet {
```

Comment:
Document what is ParamLet, and how it can be used

Reply:
removed for now